src/engraving/libmscore/excerpt.cpp

```diff
@@ -109,6 +109,22 @@ bool Excerpt::isEmpty() const
     return partScore() ? partScore()->parts().empty() : true;
 }
 
+void Excerpt::removePart(const QString& id)
+{
+    int index = 0;
+    for (Part* part: parts()) {
+        if (part->id() == id) {
+            break;
+        }
+        ++index;
+    }
+    if (index >= _parts.size()) {
+        return;
+    }
```

Comment:
Maybe cleaner as.
```
auto iter= std::find_if(parts().begin(), parts().end(), [id](auto p) { return p->id() == id; }); 
if (iter != parts.end())
  partScore()->undoRemovePart(partScore()->parts().at(std::distance(parts.begin(), iter)));
```
?
But can't you walk the parts in partScore()->parts() to find the one with the matching id?  Relying on indices matching up between two collections is always fraught!